From 5c0bbe857c265b889f2366badc03198712837131 Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Wed, 19 Aug 2020 14:49:35 -0400
Subject: [PATCH 623/767] drivers: marvell: otx2-sdei-ghes: correct issues with
 crashdump kernel

commit 1e0038e55825642bcd2918b160cb7126a7a4d163 from
git@git.assembla.com:cavium/WindRiver.linux.git

Preclude operation under crash-dump kernel as the crash-dump
environment is intended to be a minimal, stable, recovery environment;
this driver is not needed nor desired.
Also, the kernel memory mapping is not compatible with the ACPI GHES
tables used.

Change-Id: I1ee63dfb1f6428266960bce3e2e1537ac8dbf04b
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/34244
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
---
 drivers/soc/marvell/Kconfig                         |  1 +
 drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c | 10 ++++++++++
 2 files changed, 11 insertions(+)

diff --git a/drivers/soc/marvell/Kconfig b/drivers/soc/marvell/Kconfig
index 8c9c51097d2e..44fb6278355a 100644
--- a/drivers/soc/marvell/Kconfig
+++ b/drivers/soc/marvell/Kconfig
@@ -88,6 +88,7 @@ config OCTEONTX2_SDEI_GHES
 	select ARM_SDE_INTERFACE
 	select ACPI_APEI
 	select ACPI_APEI_GHES
+	select CRASH_DUMP
 	help
 	  Select this option to enable support for RAS Generic Hardware Error
 	  Source (GHES) reporting.
diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
index 35212a0cb2aa..b65e346400ec 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
@@ -21,6 +21,7 @@
 #include <linux/acpi.h>
 #include <acpi/apei.h>
 #include <linux/pci.h>
+#include <linux/crash_dump.h>
 #include "otx2-sdei-ghes.h"
 
 #define DRV_NAME       "sdei-ghes"
@@ -825,6 +826,15 @@ static int __init sdei_ghes_driver_init(void)
 
 	rc = -ENODEV;
 
+	/* Operation under kdump crash kernel is not desired nor supported. */
+#ifdef CONFIG_CRASH_DUMP
+	if (is_kdump_kernel())
+#else
+#pragma message "CONFIG_CRASH_DUMP setting is rquired for this module"
+	if (true)
+#endif
+		return rc;
+
 #ifdef CONFIG_OCTEONTX2_SDEI_GHES_BERT
 	of_node = of_find_matching_node_and_match(NULL, bed_bert_of_match,
 						  NULL);
-- 
2.31.1

